addons/l10n_it_edi/tests/test_ir_mail_server.py

```diff
@@ -82,13 +84,118 @@ def setUpClass(cls):
+            'invoice_date': datetime.date(2022, 3, 24),
+            'partner_id': cls.italian_partner_a.id,
+            'invoice_line_ids': [
+                (0, 0, {
+                    **cls.standard_line,
+                    'name': 'something price included',
+                    'tax_ids': [(6, 0, [cls.price_included_tax.id])]
+                }),
+                (0, 0, {
+                    **cls.standard_line,
+                    'name': 'something else price included',
+                    'tax_ids': [(6, 0, [cls.price_included_tax.id])]
+                }),
+                (0, 0, {
+                    **cls.standard_line,
+                    'name': 'something not price included',
+                }),
+            ],
+        })
+
+        cls.partial_discount_invoice = cls.env['account.move'].with_company(cls.company).create({
+            'move_type': 'out_invoice',
+            'invoice_date': datetime.date(2022, 3, 24),
+            'partner_id': cls.italian_partner_a.id,
+            'invoice_line_ids': [
+                (0, 0, {
+                    **cls.standard_line,
+                    'name': 'no discount',
+                }),
+                (0, 0, {
+                    **cls.standard_line,
+                    'name': 'special discount',
+                    'discount': 50,
+                }),
+                (0, 0, {
+                    **cls.standard_line,
+                    'name': "an offer you can't refuse",
+                    'discount': 100,
+                }),
+            ],
+        })
+
+        cls.full_discount_invoice = cls.env['account.move'].with_company(cls.company).create({
+            'move_type': 'out_invoice',
+            'invoice_date': datetime.date(2022, 3, 24),
+            'partner_id': cls.italian_partner_a.id,
+            'invoice_line_ids': [
+                (0, 0, {
+                    **cls.standard_line,
+                    'name': 'nothing shady just a gift for my friend',
+                    'discount': 100,
+                }),
+            ],
+        })
+        # post the invoices
+        cls.price_included_invoice._post()
+        cls.partial_discount_invoice._post()
+        cls.full_discount_invoice._post()
+
+        cls.test_invoice_xmls = {k: cls._get_test_file_content(v) for k, v in [
+            ('normal_1', 'IT01234567890_FPR01.xml'),
+            ('signed', 'IT01234567890_FPR01.xml.p7m'),
+            ('price_included', 'IT00470550013_price_included.xml'),
+            ('discounted_mixed', 'IT00470550013_discounted_mixed.xml'),
+            ('discounted_fully', 'IT00470550013_discounted_fully.xml'),
+        ]}
+
     @classmethod
     def _get_test_file_content(cls, filename):
         """ Get the content of a test file inside this module """
         path = 'l10n_it_edi/tests/expected_xmls/' + filename
         with tools.file_open(path, mode='rb') as test_file:
             return test_file.read()
 
+    def _render_invoice(self, invoice):
```

Comment:
There should be other methods you can call to the contents.  _get_invoice_edi_content or just process them but mocking a method.  (check the other localizations)